src/common-opts.ts

```diff
@@ -0,0 +1,39 @@
+import os from "os";
+import path from "path";
+
+module.exports = {
+  cache: {
+    describe: "Path to the global tink cache",
+  },
+  json: {
+    default: false,
+    describe: "Output in JSON format.",
+    type: "boolean",
+  },
+  loglevel: {
+    default: "notice",
+    alias: ["log", "l"],
+    describe: "Logger output level.",
+    choices: ["silent", "error", "warn", "http", "verbose", "info", "notice", "silly"],
+  },
+  registry: {
+    alias: "r",
+    default: "https://registry.npmjs.org",
+    describe: "Registry to ping",
+  },
+  userconfig: {
+    default: path.join(os.homedir(), ".npmrc"),
+    describe: "Path to user config file.",
+  },
+  force: {
+    type: "boolean",
+    describe: "Force an action (use with care). This should be defined per-command.",
+  },
+  log: {},
+  prefix: {},
+  then: {},
+};
+
+Object.keys(module.exports).forEach(k => {
```

Comment:
Why the mixed module types here?

Reply:
So I didn't write any of this. 😉 I just moved the `js` to `ts`. @zkat has the answers to these kinds of questions.

Reply:
Oh right on, good to know.